### PR TITLE
Cleanups in GroupAlsoByWindowEvaluatorFactory

### PR DESCRIPTION
**Pull request by tgroh**
Java8 migration, removal of an unused field, removal of an unneeded
cast.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by tgroh** [comment]
R: @huygaa11

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/GroupAlsoByWindowEvaluatorFactory.java [hunk 1]
```diff
@@ -199,12 +196,10 @@ public void processElement(WindowedValue<KeyedWorkItem<K, V>> element) throws Ex
      * Returns an {@code Iterable<WindowedValue<InputT>>} that only contains non-late input
      * elements.
      */
-    public Iterable<WindowedValue<V>> dropExpiredWindows(
+    Iterable<WindowedValue<V>> dropExpiredWindows(
```
Comment (lukecwik) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/GroupAlsoByWindowEvaluatorFactory.java]: Is there a reason why we duplicate `LateDataUtils.dropExpiredWindows`?

Comment (tgroh) [reply runners/direct-java/src/main/java/org/apache/beam/runners/direct/GroupAlsoByWindowEvaluatorFactory.java]: Combination of ordering (I believe this was introduced first) plus a minor argument conflict (we use some fields here that don't have the same type as the `Utils` method (CounterCell vs Counter being the big difference)

Comment (lukecwik) [reply runners/direct-java/src/main/java/org/apache/beam/runners/direct/GroupAlsoByWindowEvaluatorFactory.java]: Thats unfortunate.

**Pull request merged**